include/seqan3/argument_parser/detail/format_parse.hpp

```diff
@@ -339,15 +339,16 @@ class format_parse : public format_base
 
     /*!\brief Parses the given option value and appends it to the target container.
      * \tparam container_option_t Must model the seqan3::sequence_container and
-     *                            its value_type must model the seqan3::input_stream_over
+     *                            its value_type must model seqan3::input_stream_over or seqan3::named_enumeration
      *
      * \param[out] value The container that stores the parsed value.
      * \param[in] in The input argument to be parsed.
      * \returns A seqan3::option_parse_result whether parsing was successful or not.
      */
     template <sequence_container container_option_t>
     //!\cond
-        requires input_stream_over<std::istringstream, typename container_option_t::value_type>
+        requires input_stream_over<std::istringstream, typename container_option_t::value_type> ||
+                 named_enumeration<typename container_option_t::value_type>
```

Comment:
```suggestion
        requires requires (std::ranges::value_type<container_option_t> & container_value, std::string const & in)
        {
           { parse_option_value(container_value, in) } -> std::same_as<option_parse_result>;
        }
```

wouldn't that be more general? If it is a container, try to defer to its value_type (that's exactly what the function does). The only problem would be container that have themselves as value_type xD

Since we also have overloads for bool and other things, I guess `std::vector<bool>` does not work right now?

Reply:
Very good catch, `std::vector<bool>` did not work. The concept was hard to write... tldr: you need an instance of the class if you want to test member functions, but the compiler does not necessarily tell you :)

I opted to use `container_option_t::value_type` instead of the ranges thing:
1. Doesn't seem to be currently included (less includes :) )
2. It currently requires `sequence_container`, which requires `value_type` to be defined

If this is good, you can squash

Reply:
Very nice!